### PR TITLE
build: in_syslog: Enable in_syslog plugin on Windows

### LINKED ISSUE #7235: syslog input does not work on Windowss
## Bug Report

**Describe the bug**

The documentation around the syslog input does not say anything about restrictions and that it is not available on Windows. 

But if you try to run with the syslog input on Windows, it fails with the message `[error] [config] section 'syslog' tried to instance a plugin name that don't exists`. The issue [5153 ](https://github.com/fluent/fluent-bit/issues/5153) goes into the reason to why this happens. But it does not answer the question if this is intentional, and whether there is a plan to fix it.

Furthermore, the issue [960 Add Windows Support to Fluent Bit](https://github.com/fluent/fluent-bit/issues/960) lists `in_syslog `as one of the high priority plugins that would be ported, and no mention that it was not.

**To Reproduce**
Configure fluent-bit to run with the syslog input on a Windows machine.

**Expected behavior**
syslog either working, or it being documented that it does not, with a strong preference to the former.

**Your Environment**

* Version used: 2.0.9
* Configuration:
[INPUT]
   Name              syslog
   Parser            syslog-rfc3164
   Mode              tcp
   Listen            0.0.0.0
   Port              24225
   Buffer_Chunk_Size 32KB
   Buffer_Max_Size   64KB
* Operating System and version: Windows Server 2019
* Filters and plugins: syslog

**Additional context**

syslog is used for [ATNA](https://profiles.ihe.net/ITI/TF/Volume1/ch-9.html) which is used in Windows-heavy environments.

### PR DESCRIPTION
**Pull request by cosmo0920**
Closes #7235
Since Windows does not have unix socket, we have to disable on that platform.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: plugins/in_syslog/syslog_server.c [hunk 1]
```diff
@@ -123,6 +126,7 @@ static int syslog_server_unix_create(struct flb_syslog *ctx)
 
     return 0;
 }
+#endif
```
Comment (leonardo-albertovich) [inline plugins/in_syslog/syslog_server.c:129]: This conditional is fine but I need you to add this before the `#endif` : 

```
#else
static int syslog_server_unix_create(struct flb_syslog *ctx)
{
    return -1;
}
```

**Review by leonardo-albertovich — CHANGES_REQUESTED** [review]
Let's take a step back and approach this from a different point of view to keep the modifications and conditional code to a minimal degree.

As a first step I need you to remove all the `#if !defined(FLB_SYSTEM_WINDOWS)` conditionals.

Then in `syslog_conf_create`  inside [this](https://github.com/fluent/fluent-bit/blob/4ced1e865541dedbe0d422f79fd955359dc31246/plugins/in_syslog/syslog_conf.c#L72) conditional as a first step I need you to to add a block like this : 

```
#ifdef FLB_SYSTEM_WINDOWS
if (strcasestr(ctx->mode_str, "unix") != NULL) {
        flb_log_event_encoder_destroy(ctx->log_encoder);

        flb_plg_error(ins, "unix sockets are note available in windows");
        flb_free(ctx);

        return NULL;
}

#undef FLB_SYSLOG_UNIX_UDP
#define FLB_SYSLOG_UNIX_UDP FLB_SYSLOG_UDP

#endif
``` 

Which would cause the remaining code (including the fallback branch to properly set the socket type in windows.

The idea is to only wrap the parts that break compilation in conditional preprocessor directives (and in the case of the function then just define an alternative one that unconditionally fails).

That way we can have the same result with only 2 preprocessor conditionals and a much cleaner and homogeneous codebase.

File: plugins/in_syslog/syslog_server.c (hunk 1 above)
Comment (cosmo0920) [reply plugins/in_syslog/syslog_server.c:134]: I like your suggestion. I applied that change.

**Comment by cosmo0920** [comment]
> That way we can have the same result with only 2 preprocessor conditionals and a much cleaner and homogeneous codebase.

However, we need to use three conditions for supporting Windows. Adjusting to Include headers also needed to support Windows. Unix socket related headers are not provided on that platform.

**Comment by patrick-stephens** [comment]
I've triggered a test build of all supported targets just to confirm there are no issues.

**Pull request merged**